### PR TITLE
ENH: add suuport for slicing an `IndelMap` with a step > 1 

### PR DESCRIPTION
**Pull request by KatherineCaley**
mods in this PR: 

`location.py`
- support for slicing `IndelMap` object with a positive stride, will work on negative strides now that I've worked out the logic for positive strides, partially addresses #1967

### CODE REVIEW

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Thanks for this!
Couple of suggestions, plus I suggest some integration tests, i.e. with an Aligned instance, slicing with stride matches same operation on the underlying string.

File: tests/test_core/test_location.py [hunk 1]
```diff
@@ -1195,3 +1196,44 @@ def test_indelmap_make_seq_feature_map():
...
+        slice(4, 16, 2),
+        slice(1, 16, 7),
+    ],
+)
+@pytest.mark.parametrize(
+    "data",
+    [
+        "--AAA---C--AAAA--",
+        "AAACCCCTTTTGGGGAA",
+        "--AAA---C--AAAAAA",
```
[the remark this answers is not in the document]
Comment (KatherineCaley) [reply tests/test_core/test_location.py:1227]: good point!

**Comment by KatherineCaley** [comment]
@GavinHuttley  I suggest that I add integration with `Aligned` in the next PR, as I have a reasonable amount of stashes changed for Aligned that I've been trying to keep separate, mainly
- implementing changes to the design of aligned/alignment as discussed on 30/07 (see board pic in zulip)
- removing the index tracking capabilities from AlignedDataView and trying to mirror the SeqsData design more closely in AlignedSeqsData

happy to push everything up now if you'd prefer 😊

**Review by GavinHuttley — APPROVED** [review]
Bonza bonza @KatherineCaley !

**Pull request merged**